### PR TITLE
Prevent merging PRs which have CI checks run a week+ ago

### PR DESCRIPTION
**Pull request by mkArtakMSFT**

### CODE REVIEW

**Review by wtgodbe — APPROVED** [review]
LGTM - I asked other folks on the team for feedback, so let's wait a bit to see if anybody has other thoughts

**Review by wtgodbe — CHANGES_REQUESTED** [review]
Actually, can we change the `approve` action to only approve if somebody else has already approved the PR? That way we don't accidentally merge any PRs that have only been reviewed by fabricbot. Or just remove the rejection without approving?

**Comment by mkArtakMSFT** [comment]
> can we change the approve action to only approve if somebody else has already approved the PR?

Can't find anything relevant :( 
image

**Comment by wtgodbe** [comment]
Hmm, maybe `lock` when adding the label and `unlock` when removing then?

**Comment by HaoK** [comment]
Could we instead just have the bot do /azp run on old green PRs?

File: .github/fabricbot.json
Comment (dougbu) [inline .github/fabricbot.json:3609]:
```suggestion
              "days": 7
```
❔ (not sure if this parameter is counted oddly)

**Comment by dougbu** [comment]
> Could we instead just have the bot do /azp run on old green non-draft PRs?

We discussed that option but realized aspnetcore alone has many PRs that would get an `/azp run` in them, which may have a significant resource impact. This way the runs happen only when the owner or one of the repo collaborators feels the PR is actually ready to go.

On the other hand, there's a related point about servicing PRs: Because those often sit while the branches are closed and very rarely hit breaking changes from other PRs, might be good to focus on 'main' w/ this new action.

**Comment by mkArtakMSFT** [comment]
Instead of approving the PR, maybe we should not do any of it, and instead just automate further by dropping an `/azp run` when the `pending-ci-rerun` label is removed?

File: .github/fabricbot.json
Comment (dougbu) [inline .github/fabricbot.json:3595]: Should this rule also confirm the PR has been approved❔ Not liking scenarios in which this rule runs when nobody has approved a PR, the PR owner or another collaborator runs the pipeline again, the `Unblock PRs which are no longer awaiting for a ci run` rule runs, and the PR is merged because it's "approved" and runs smoothly.

**Review by dougbu — COMMENTED** [review]
I suspect this is fine but have one concern&hellip;

File: .github/fabricbot.json
Comment (mkArtakMSFT) [reply .github/fabricbot.json:3595]: Well, my last comment was addressing exactly this - "instead of approving the PR"

Comment (dougbu) [reply .github/fabricbot.json:3595]: Agreed. I hadn't seen your comment when I made mine.

Re
> instead just automate further by dropping an /azp run when the pending-ci-rerun label is removed?

That has the downside of potentially rerunning a build that just completed. We aren't telling people to remove the label but to rerun. It could work if you change the comment language in the rule where we add the label.

The other option I considered but now think is likely worse would be to "confirm the PR has been approved" in the search terms. The downside there is the tendency to merge as soon as approvals come through, leaving the bot reacting too late.

Thinking further out of the box, can the bot withdraw its rejection w/o approving the PR❔

Or, should we just label the issue, reject it, and leave it to assigned person to handle removing / ignoring the bot's rejection❔ That is, limit this change to one rule❔

Comment (mkArtakMSFT) [reply .github/fabricbot.json:3588]: Updated the PR with this proposal

**Review by dougbu — APPROVED** [review]

**Comment by mkArtakMSFT** [comment]
@wtgodbe waiting for your feedback here.

**Review by wtgodbe — APPROVED** [review]
Works for me

**Comment by Nick-Stanton** [comment]
@mkArtakMSFT @wtgodbe @dougbu 

image

Encountered this [here](https://github.com/dotnet/aspnetcore/pull/43928#issuecomment-1406966504)

**Pull request merged**
